app/services/BackgroundTaskService.ts

```diff
@@ -21,11 +26,29 @@ class BackgroundTaskService {
         requiresStorageNotLow: false, // Default
       },
       async (taskId) => {
+        PushNotification.localNotification({
+          title: 'Running background task',
+          message: new Date().toLocaleString(),
+        });
         console.log('[js] Received background-fetch event: ', taskId);
-        IntersectService.checkIntersect(null, false);
+
+        if (Platform.OS === 'ios') {
+          NativeModules.SecureStorageManager.trimLocations();
+        }
+
+        const lastCheckedMs = Number(await GetStoreData(LAST_CHECKED));
+        if (dayjs().diff(lastCheckedMs, 'hour') >= 12) {
+          IntersectService.checkIntersect(null);
+          await SetStoreData(LAST_CHECKED, String(dayjs().valueOf()));
+        }
```

Comment:
Can we move the 'skip' console log to here now? I think testing might be using that for ensuring runs, so that'll be good to keep consistent